common/persistence/elasticsearch/visibility_store.go

```diff
@@ -1197,3 +1197,24 @@ func parseHHMMSSDuration(d string) (int64, error) {
 
 	return hours*int64(time.Hour) + minutes*int64(time.Minute) + seconds*int64(time.Second) + nanos, nil
 }
+
+func detailedErrorMessage(err error) string {
+	elasticErr, isElasticErr := err.(*elastic.Error)
```

Comment:
Can you make this more future-proof with `errors.As`?

Reply:
> errors.As

what will errors.As provide? if not more complexity?

Reply:
It is canonical go way to check error type.